spaceapi/src/lib.rs

```diff
@@ -8,11 +8,35 @@ use rustc_serialize::json::{Json, ToJson};
 /// An Optional value can contain Optional::Some<T> or Optional::Absent.
 /// It is similar to an Option, but Optional::Absent means it will be
 /// omitted when serialized.
+#[derive(Debug, Copy, Clone)]
 pub enum Optional<T> {
     Value(T),
     Absent,
 }
 
+impl<T> Optional<T> {
+    pub fn map_or<U, F: FnOnce(T) -> U>(self, def: U, f: F) -> U {
+        match self {
+            Optional::Value(v) => f(v),
+            Optional::Absent => def,
+        }
+    }
```

Comment:
My proposal was not to serialise to `null` in the end but only in a first step. This is what I would expect the API to accept. But since the API doesn't support it, I'd filter out the `null`s, replace them with undefined and let `ToJson` do the hard work.